src/util/irep_ids.def

```diff
@@ -217,6 +217,7 @@ IREP_ID_TWO(cpp_new_array, cpp_new[])
 IREP_ID_TWO(cpp_delete_array, cpp_delete[])
 IREP_ID_ONE(java_new)
 IREP_ID_ONE(java_new_array)
```

Comment:
Why is java_new_array still needed?

Reply:
Previously `ID_java_new_array` was being used to denote **both** the new-array-structure operation (yields something like `{int length; void** data;}*`) **and** the new-plain-array op (which yields a `void**`, like a C++ `new[]`). It was getting away with it because of the order `builtin_functions.cpp` looked for the different cases, but it was very fragile in case that changed. This way the two ops have distinct names (new_array and new_array_data respectively)